test/images/agnhost/serve-hostname/serve_hostname.go

```diff
@@ -27,19 +28,38 @@ import (
 	"os/signal"
 	"syscall"
 	"time"
+
+	"github.com/spf13/cobra"
 )
 
+// CmdServeHostname is used by agnhost Cobra.
+var CmdServeHostname = &cobra.Command{
+	Use:   "serve-hostname",
+	Short: "Serves the hostname",
+	Long:  `Serves the hostname through HTTP / TCP / UDP on the given port.`,
+	Args:  cobra.MaximumNArgs(4),
+	Run:   main,
+}
+
 var (
-	doTCP   = flag.Bool("tcp", false, "Serve raw over TCP.")
-	doUDP   = flag.Bool("udp", false, "Serve raw over UDP.")
-	doHTTP  = flag.Bool("http", true, "Serve HTTP.")
-	doClose = flag.Bool("close", false, "Close connection per each HTTP request")
-	port    = flag.Int("port", 9376, "Port number.")
+	doTCP   bool
+	doUDP   bool
+	doHTTP  bool
+	doClose bool
+	port    int
 )
 
-func main() {
+func init() {
+	CmdServeHostname.Flags().BoolVar(&doTCP, "tcp", false, "Serve raw over TCP.")
```

Comment:
Did you test that the flag code refactoring went smoothly? (E.g., did you try to run the program with the flag arguments?)

Also, IIRC the flag package uses single-dash flags; I'm not sure how Cobra flags are like --- but hopefully it also uses single-dash flags for compatibility with the old one (to not surprise any existing users). If not, this change should be mentioned somewhere.

Reply:
I haven't manually tested every single subcommand and flag, but this refactor is the same as on the other parts and those worked fine.

Cobra flags are using double dashes (--). It only supports a single dash for 1 character flags though. The image replacement PR will also update those things as well. Unfortunately, this will have to be updated if anyone else is using the Kubernetes test images, and I suppose we could mention this in the README file.